ui/src/containers/AlertProvider.js

```diff
@@ -22,7 +22,8 @@ export function useAlerts(filters: FilterLabels) {
   return query;
 }
 
-const AlertProvider = ({ children }: any) => {
+// the `fakequery` is for testing purpose in order to inject alerts
+const AlertProvider = ({ children, fakequery }: any) => {
```

Comment:
```suggestion
import type {Node} from "react";
const AlertProvider = ({ children }: {children: Node}) => {
```

Instead of introducing a `fakeQuery` here I would go for adding an optional second parameter to useAlerts, this optional second parameter would be the alert context. Doing so in the test you won't need to render an AlertProvider wrapper anymore and could `createContext` initialised with your mocked query.

Reply:
Another alternative (maybe better because in this way it won't impact the definition of useAlerts neither useHighestSeverityAlerts definition) could be to move the context to a separate module, mock with jest.mock and returning a context containing your fakeQuery

Reply:
Sure. thanks for the suggestion. Will go the second solution!